jetty-websocket/javax-websocket-client/src/main/java/org/eclipse/jetty/websocket/javax/client/JavaxWebSocketClientContainer.java

```diff
@@ -52,33 +53,29 @@
 @ManagedObject("JSR356 Client Container")
 public class JavaxWebSocketClientContainer extends JavaxWebSocketContainer implements javax.websocket.WebSocketContainer
 {
-    protected final WebSocketCoreClient coreClient;
+    protected WebSocketCoreClient coreClient;
+    protected Supplier<WebSocketCoreClient> coreClientFactory;
     private final JavaxWebSocketClientFrameHandlerFactory frameHandlerFactory;
-    private ClassLoader contextClassLoader;
     private DecoratedObjectFactory objectFactory;
     private WebSocketExtensionRegistry extensionRegistry;
 
     public JavaxWebSocketClientContainer()
     {
-        this(new WebSocketCoreClient());
-        this.coreClient.getHttpClient().setName("Javax-WebSocketClient@" + Integer.toHexString(this.coreClient.getHttpClient().hashCode()));
-        // We created WebSocketCoreClient, let lifecycle be managed by us
-        addManaged(coreClient);
+        this(() -> new WebSocketCoreClient());
     }
 
-    public JavaxWebSocketClientContainer(HttpClient httpClient)
+    public JavaxWebSocketClientContainer(Supplier<WebSocketCoreClient> coreClientFactory)
     {
-        this(new WebSocketCoreClient(httpClient));
-        // We created WebSocketCoreClient, let lifecycle be managed by us
-        addManaged(coreClient);
+        this((WebSocketCoreClient)null);
```

Comment:
This lacks the knowledge of who created the WebSocketCoreClient.

Reply:
The factory will always create or provide it in this case.  Hence I agree the factory should do the naming